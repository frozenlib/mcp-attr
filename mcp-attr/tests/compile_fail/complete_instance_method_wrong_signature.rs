use mcp_attr::{server::{mcp_server, McpServer}, Result};

struct TestServer;

impl TestServer {
    async fn wrong_signature_method(&self, wrong_param: i32) -> Result<Vec<String>> {
        Ok(vec![wrong_param.to_string()])
    }
}

#[mcp_server]
impl McpServer for TestServer {
    #[prompt]
    async fn test_prompt(&self, #[complete(.wrong_signature_method)] msg: String) -> Result<String> {
        Ok(msg)
    }
}

fn main() {}