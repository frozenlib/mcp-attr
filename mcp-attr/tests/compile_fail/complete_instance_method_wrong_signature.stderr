error[E0407]: method `wrong_signature_method` is not a member of trait `McpServer`
  --> tests/compile_fail/complete_instance_method_wrong_signature.rs:12:5
   |
12 | /     async fn wrong_signature_method(&self, wrong_param: i32) -> Result<Vec<String>> {
13 | |         Ok(vec![wrong_param.to_string()])
14 | |     }
   | |_____^ not a member of trait `McpServer`

error[E0599]: no method named `wrong_signature_method` found for reference `&TestServer` in the current scope
 --> tests/compile_fail/complete_instance_method_wrong_signature.rs:8:45
  |
8 |     async fn test_prompt(&self, #[complete(.wrong_signature_method)] msg: String) -> Result<String> {
  |                                             ^^^^^^^^^^^^^^^^^^^^^^ method not found in `&TestServer`
